freqtrade/templates/base_strategy.py.j2

```diff
@@ -37,6 +37,9 @@ class {{ strategy }}(IStrategy):
     # Check the documentation or the Sample strategy to get the latest version.
     INTERFACE_VERSION = 2
 
+    trading_mode: "{{ trading_mode }}"
+    margin_mode: "{{ margin_mode }}"
```

Comment:
these arguments can't be set from the strategy (and i think that makes sense to keep it this way).

You'll have to have the exchange configured correctly (like - amounts in the futures wallet, API keys enabled for futures, ...) - so you have to have the config set correctly.

Reply:
Yes, should leave this to the config file.